src/MoBi.Presentation/Tasks/MoBiFormulaTask.cs

```diff
@@ -175,28 +232,29 @@ public IFormula CreateNewFormula(Type formulaType, IDimension formulaDimension)
 
       public TFormula CreateNewFormula<TFormula>(IDimension formulaDimension) where TFormula : IFormula
       {
-         return CreateNewFormula(typeof (TFormula), formulaDimension).DowncastTo<TFormula>();
+         return CreateNewFormula(typeof(TFormula), formulaDimension).DowncastTo<TFormula>();
       }
 
       public IMoBiCommand UpdateDistributedFormula(IDistributedParameter distributedParameter, IDistributionFormula newDistributedFormula, string formulaType, IBuildingBlock buildingBlock)
       {
          _context.Register(newDistributedFormula);
          _context.Register(distributedParameter.Formula);
-         return new UpdateDistributedFormulaCommand(distributedParameter, newDistributedFormula, formulaType, buildingBlock).Run(_context);
+         var command = new UpdateDistributedFormulaCommand(distributedParameter, newDistributedFormula, formulaType, buildingBlock).Run(_context);
+         return withUpdatedDefaultStateAndValueOrigin(command, newDistributedFormula, buildingBlock);
       }
 
       private IFormula createFormulaFromType(Type formulaType)
       {
-         if (formulaType == typeof (ConstantFormula))
+         if (formulaType == typeof(ConstantFormula))
             return _context.Create<ConstantFormula>();
 
-         if (formulaType == typeof (BlackBoxFormula))
+         if (formulaType == typeof(BlackBoxFormula))
             return _context.Create<BlackBoxFormula>();
 
-         if (formulaType == typeof (TableFormula))
+         if (formulaType == typeof(TableFormula))
             return createTableFormula();
 
-         if (formulaType == typeof (SumFormula))
+         if (formulaType == typeof(SumFormula))
             return _context.Create<SumFormula>();
 
          if (formulaType == typeof(TableFormulaWithOffset))
```

Comment:
case `DistributionFormula `not required here?

Reply:
That's the old code here . Nothing to do with the PR